### PR TITLE
Add on_unimplemented note to Index

### PR DESCRIPTION
**Pull request by Manishearth**
Helps issues like [these](http://www.reddit.com/r/rust/comments/2tpefm/unable_to_access_array_elements/)

r? @Gankro

rollup-worthy

### CODE REVIEW

**Comment by Gankro** [comment]
FOR REALS THIS TIME

File: src/libcore/ops.rs [hunk 1]
```diff
@@ -895,6 +895,7 @@ shr_impl_all! { u8 u16 u32 u64 usize i8 i16 i32 i64 isize }
 /// }
 /// ```
 #[lang="index"]
+#[rustc_on_unimplemented = "the type {Self} cannot be indexed by {Index}"]
```
Comment (P1start) [inline src/libcore/ops.rs]: Hmm, there should probably be backticks around `{Self}` and `{Index}` here.

**Comment by Manishearth** [comment]
@Gankro Okay, perhaps not for reals? :> Need another r+ :p

**Comment by Gankro** [comment]
What changed?

**Comment by Manishearth** [comment]
Backticks were added.

**Pull request merged**